tests/tracking/request_header/test_registry.py

```diff
@@ -0,0 +1,160 @@
+    DatabricksRequestHeaderProvider,
+)
+
+# pylint: disable=unused-argument
+
+def test_request_header_context_provider_registry_register():
+    provider_class = mock.Mock()
+
+    registry = RequestHeaderProviderRegistry()
+    registry.register(provider_class)
+
+    assert set(registry) == {provider_class.return_value}
+
+
+def test_request_header_provider_registry_register_entrypoints():
+    provider_class = mock.Mock()
+    mock_entrypoint = mock.Mock()
+    mock_entrypoint.load.return_value = provider_class
+
+    with mock.patch(
+        "entrypoints.get_group_all", return_value=[mock_entrypoint]
+    ) as mock_get_group_all:
+        registry = RequestHeaderProviderRegistry()
+        registry.register_entrypoints()
+
+    assert set(registry) == {provider_class.return_value}
+    mock_entrypoint.load.assert_called_once_with()
+    mock_get_group_all.assert_called_once_with("mlflow.request_header_provider")
+
+
+@pytest.mark.parametrize(
+    "exception", [AttributeError("test exception"), ImportError("test exception")]
+)
+def test_request_header_provider_registry_register_entrypoints_handles_exception(exception):
+    mock_entrypoint = mock.Mock()
+    mock_entrypoint.load.side_effect = exception
+
+    with mock.patch(
+        "entrypoints.get_group_all", return_value=[mock_entrypoint]
+    ) as mock_get_group_all:
+        registry = RequestHeaderProviderRegistry()
+        # Check that the raised warning contains the message from the original exception
+        with pytest.warns(UserWarning, match="test exception"):
+            registry.register_entrypoints()
+
+    mock_entrypoint.load.assert_called_once_with()
+    mock_get_group_all.assert_called_once_with("mlflow.request_header_provider")
+
+
+def _currently_registered_request_header_provider_classes():
+    return {
+        provider.__class__
+        for provider in mlflow.tracking.request_header.registry._request_header_provider_registry
+    }
+
+
+def test_registry_instance_defaults():
+    expected_classes = {DatabricksRequestHeaderProvider}
+    assert expected_classes.issubset(_currently_registered_request_header_provider_classes())
+
+
+def test_registry_instance_loads_entrypoints():
+    class MockRequestHeaderProvider(object):
+        pass
+
+    mock_entrypoint = mock.Mock()
+    mock_entrypoint.load.return_value = MockRequestHeaderProvider
+
+    with mock.patch(
+        "entrypoints.get_group_all", return_value=[mock_entrypoint]
+    ) as mock_get_group_all:
+        # Entrypoints are registered at import time, so we need to reload the module to register the
+        # entrypoint given by the mocked extrypoints.get_group_all
+        reload(mlflow.tracking.request_header.registry)
+
+    assert MockRequestHeaderProvider in _currently_registered_request_header_provider_classes()
+    mock_get_group_all.assert_called_once_with("mlflow.request_header_provider")
+
+
+@pytest.mark.large
+def test_run_context_provider_registry_with_installed_plugin(tmp_wkdir):
```

Comment:
what does `tmp_wkdir` do for us here?

Reply:
to be honest, i dont actually know. I copied it from `run_context_provider`. When I take it out the test still seems to pass. I think it might be something to do with the test plugin but I'm not sure.